roles/docker/tasks/systemcontainer_crio.yml

```diff
@@ -3,16 +3,16 @@
 # TODO: Much of this file is shared with container engine tasks
 - set_fact:
     l_insecure_crio_registries: "{{ '\"{}\"'.format('\", \"'.join(l2_docker_insecure_registries)) }}"
-  when: l2_docker_insecure_registries
+  when: l2_docker_insecure_registries | bool
 - set_fact:
     l_crio_registries: "{{ l2_docker_additional_registries + ['docker.io'] }}"
-  when: l2_docker_additional_registries
+  when: l2_docker_additional_registries | bool
 - set_fact:
     l_crio_registries: "{{ ['docker.io'] }}"
-  when: not l2_docker_additional_registries
+  when: not (l2_docker_additional_registries | bool)
 - set_fact:
     l_additional_crio_registries: "{{ '\"{}\"'.format('\", \"'.join(l_crio_registries)) }}"
-  when: l2_docker_additional_registries
+  when: l2_docker_additional_registries | bool
```

Comment:
Note: I'm not aware of the standards or context here much at all.

FWIW:  In my own struggles with the ``bool`` filter,  I found Ansible (despite being written in python) does a really poor job of type-converting values for logic testing.  That filter, literally **only** handles booleans, and strings of "True", "Yes", or "1" - everything else is considered False 😭 

So I came up with this little gem, FWIW:

``when: some_var | default("",True) | trim | length != 0``

It takes care of undefined variables, and will convert just about everything (empty strings, empty lists, hashes, booleans, integers, etc.) into a boolean-testable value.  The only thing it doesn't handle well is the "False" string.  However that's easily tested separately using the ``bool`` filter.

Reply:
We aren't checking for `undefined` variables here.  We are ensuring the result doesn't include `{{ }}'s`.  If you `|bool` then it removes the underlying variable's `{{`'s which ansible complains about as they are deprecated in `when` statements.